### PR TITLE
fix(core):  scope selectors inside a functional pseudo-class correctly

### LINKED ISSUE #1968: any none root selectors after the first in a functional pseudo-class is not transformed correctly
### Actual Behavior

[Playground link](https://wixplosives.github.io/stylable-playground/#state=%28%27files%21%28*%21%27.a%2520%280%2520%2520%2520%2520-st-states%253A%2520b%252Cc%253B0%2900.a%253Ais%257B%253Ab%252C%253Ac%257D%28%29%27%29%7Eselected%21*%27%29*%27%252Findex.st.css0%255Cn%25010*_)

After the first state inside the `:is`, it does not transform properly.
(It resets the anchor to the root)

### Expected Behavior

After each iteration on the list of chunks, it should reset to the last context and not to the root.
https://github.com/wix/stylable/blob/master/packages/core/src/stylable-transformer.ts#L484

### Version

4.5.0

### Relevant log output

_No response_

### Additional Context

_No response_

### PR DESCRIPTION
**Pull request by tzachbon**
Closes #1968

### CODE REVIEW

File: packages/core/test/pseudo-states.spec.ts [hunk 1]
```diff
@@ -2027,6 +2027,34 @@ describe('pseudo-states', () => {
                 });
             });
         });
+
+        describe('state inside selector list', () => {
```
Comment (idoros) [inline packages/core/test/pseudo-states.spec.ts:2031]: I know I said that  maybe `pseudo-states.spec` is a good place for this, but I mislead you. this spec is for stylable custom states. this is simply a transform of functional pseudo-classes. There is no specific spec file for pseudo-classes, so maybe add a describe for `pseudo-classes` in `scoping.spec` and place it under there for now.

File: packages/core/test/pseudo-states.spec.ts [hunk 2]
```diff
@@ -2027,6 +2027,34 @@ describe('pseudo-states', () => {
...
+                        },
+                    },
+                });
+
+                expect(
+                    result.meta.diagnostics.reports,
+                    'no diagnostics reported for imported states'
+                ).to.eql([]);
+                expect(result).to.have.styleRules({
+                    1: '.entry__a:is(.entry--b, .entry--c) {}',
```
Comment (idoros) [inline packages/core/test/pseudo-states.spec.ts:2054]: use `testInlineExpects` to check for transformation results

File: packages/core/test/stylable-transformer/scoping.spec.ts [hunk 3]
```diff
@@ -799,6 +799,36 @@ describe('Stylable postcss transform (Scoping)', () => {
 
             testInlineExpects(result);
         });
+
+        it('should scope multiple pseudo-elements that extended none root anchor', () => {
```
Comment (idoros) [inline packages/core/test/stylable-transformer/scoping.spec.ts:803]: this test is redundant in my opinion. You already checked the transform context target flow in the other test. it is not specifically related to pseudo-elements

**Review by idoros — CHANGES_REQUESTED** [review]

File: packages/core/test/stylable-transformer/scoping.spec.ts (hunk 3 above)
Comment (idoros) [reply packages/core/test/stylable-transformer/scoping.spec.ts:803]: the fix is for the transformation flow of pseudo-classes, the only flow that calls `scopeSelectorAst` with a nested context is a functional pseudo-class

**Review by idoros — APPROVED** [review]

**Review by barak007 — APPROVED** [review]

**Pull request merged**